packages/vision/src/index.js

```diff
@@ -410,7 +412,9 @@ Vision.prototype.detect = function(images, options, callback) {
 
       var detections = foundImages
         .map(groupDetectionsByImage)
-        .map(removeExtraneousAnnotationObjects)
+        .map(function(image) {
+          return curation.removeExtraneousAnnotations(image, options.types);
```

Comment:
It starts the chain with `foundImages` because we're ultimately giving back a result for each image.

So, `foundImages` could be 2 images. Its values are completely ignored, because in the first step of the `map` transformation ([`groupDetectionsByImage`](https://github.com/GoogleCloudPlatform/google-cloud-node/blob/688ac58e72f950c64aac7999b57c247a0db793f5/packages/vision/src/index.js#L433)), we return from that the apiResponse, *split* into as many annotation types that were requested based on how many images.

So it's basically only used as a way to split the annotations the right amount of times. More obvious solutions are very welcome.